File: 33-file-cat/file_cat.ml
(* Ocaml program to print file contents *)
(* Written by M.V.Harish Kumar on 01-06-2024 *)
let filename = Sys.argv.(1)
let border = String.make 20 '-'

let () =
  let file = open_in filename in
    let ftxt = really_input_string file (in_channel_length file) in
      close_in file;
      Printf.printf "Contents of %s:\n" filename;
      print_endline border;
      print_string ftxt;
      print_newline ();
      print_endline border

File: 33-file-cat/file_cat.go
// Go program to print file contents
// Written by M.V.Harish Kumar on 18-06-2025
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	fname := os.Args[1]
	ftxt, _ := os.ReadFile(fname)
	border := strings.Repeat("-", 20)

	fmt.Println("Contents of", fname)
	fmt.Println(border)
	fmt.Println(string(ftxt))
	fmt.Println(border)
}
